api/tests/integration/domain/usecases/create-badge_test.js

```diff
@@ -0,0 +1,209 @@
+const _ = require('lodash');
+
+const { expect, databaseBuilder, mockLearningContent, knex, catchErr, sinon } = require('../../../test-helper');
+
+const badgeRepository = require('../../../../lib/infrastructure/repositories/badge-repository');
+const badgeCriteriaRepository = require('../../../../lib/infrastructure/repositories/badge-criteria-repository');
+const skillSetRepository = require('../../../../lib/infrastructure/repositories/skill-set-repository');
+const targetProfileRepository = require('../../../../lib/infrastructure/repositories/target-profile-repository');
+const createBadge = require('../../../../lib/domain/usecases/create-badge');
+const Badge = require('../../../../lib/domain/models/Badge');
+const { AlreadyExistingEntityError, NotFoundError, InvalidSkillSetError } = require('../../../../lib/domain/errors');
+
+describe('Integration | UseCases | create-badge', function () {
+  let targetProfileId;
+  let badge;
+  let badgeCreation;
+  let existingBadgeKey;
+  let dependencies;
+
+  beforeEach(async function () {
+    const learningContent = {
+      skills: [{ id: 'recSkill1' }],
+    };
+
+    mockLearningContent(learningContent);
+
+    targetProfileId = databaseBuilder.factory.buildTargetProfile().id;
+    learningContent.skills.forEach((skill) =>
+      databaseBuilder.factory.buildTargetProfileSkill({ targetProfileId, skillId: skill.id })
+    );
+    existingBadgeKey = databaseBuilder.factory.buildBadge().key;
+
+    await databaseBuilder.commit();
+
+    badge = {
+      key: 'TOTO23',
+      altMessage: 'example image',
+      imageUrl: 'https//images.example.net',
+      message: 'Bravo !',
+      title: 'Le super badge',
+      isCertifiable: false,
+      isAlwaysVisible: true,
+    };
+
+    badgeCreation = {
+      ...badge,
+    };
```

Comment:
```suggestion
    badgeCreation = {
      ...badge,
    };
```
Suggestion: Renommer le badge du dessus en badgeCreation (nom qui peut être également à retravailler afin de mieux retranscrire la réalité ? badgeAndCriteria ? 🤔 )